main/cron/create_course_sessions.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/* For licensing terms, see /license.txt */
+/**
+ * Create course sessions procedure. It creates sessions for courses that haven't it yet.
+ * If today is greater than OFFSET, it will create them also for the next month.
+ * @package chamilo.cron
+ * @author Imanol Losada <imanol.losada@beeznest.com>
+ */
+
+/**
+ * Initialization
+ */
+if (php_sapi_name() != 'cli') {
+    exit; //do not run from browser
+}
+
+require_once __DIR__ . "/../inc/global.inc.php";
+
+// First day of the current month to create sessions and add courses for the next month (e.g. "07")
+define("OFFSET", "15");
+
+function getMonthFirstAndLastDates($initialDate = null) {
+    $startDate = $initialDate ? $initialDate : date("Y-m-01");
+    $nextMonthStartDate = date("Y-m-d", strtotime($startDate." + 1 month"));
+    $endDate = date("Y-m-d", strtotime($nextMonthStartDate." - 1 minute"));
+    return array('startDate' => $startDate, 'endDate' => $endDate);
+}
+
+function createCourseSessions($courses, $administratorId, $startDate, $endDate) {
+    echo "\n";
+    echo $courses ?
```

Comment:
I did a little check on (bool) count(array()) and (bool) count(array(0 => false)) vs (bool) array() and I'm OK with your change. Good.